app/common/state/tabContentState/partitionState.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// State helpers
+const tabUIState = require('../tabUIState')
+const frameStateUtil = require('../../../../js/state/frameStateUtil')
+
+// Constants
+const {tabs} = require('../../../../js/constants/config')
+
+module.exports.isPartitionTab = (state, frameKey) => {
+  const frame = frameStateUtil.getFrameByKey(state, frameKey)
+
+  if (frame == null) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.error('Unable to find frame for isPartitionTab method')
+    }
+    return false
+  }
+
+  return !!frame.get('partitionNumber')
+}
+
+module.exports.getPartitionNumber = (state, frameKey) => {
+  const frame = frameStateUtil.getFrameByKey(state, frameKey)
+
+  if (frame == null) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.error('Unable to find frame for getPartitionNumber method')
+    }
+    return 0
+  }
+  const partitionNumber = frame.get('partitionNumber')
+  if (typeof partitionNumber === 'string') {
+    return partitionNumber.replace(/^partition-/i, '')
+  }
+  return partitionNumber
+}
+
+module.exports.getMaxAllowedPartitionNumber = (state, frameKey) => {
+  const frame = frameStateUtil.getFrameByKey(state, frameKey)
+
+  if (frame == null) {
+    if (process.env.NODE_ENV !== 'test') {
```

Comment:
There's a lot of these checks... can we consolidate this check into a single method? Maybe it can be added into appConfig (since that already has some environment checks)

Reply:
If there's a problem with this during unit tests, you should be able to stub `console.error` completely and have it no-op

Reply:
I'm going to remove them all. Looking now they seem silly as frames may be unavailable at start-up time and you'll end up receiving warning unnecessarily which may lead to future bug reports, where in fact there's no bug but only async methods.